org.eclipse.zest.core/src/org/eclipse/zest/core/widgets/Graph.java

```diff
@@ -160,42 +154,36 @@ public void dispatchMouseMoved(org.eclipse.swt.events.MouseEvent me) {
 		this.getLightweightSystem().getRootFigure().addMouseListener(dragSupport);
 		this.getLightweightSystem().getRootFigure().addMouseMotionListener(dragSupport);
 
-		this.nodes = new ArrayList();
+		this.nodes = new ArrayList<>();
 		this.preferredSize = new Dimension(-1, -1);
 		this.connectionStyle = ZestStyles.NONE;
 		this.nodeStyle = ZestStyles.NONE;
-		this.connections = new ArrayList();
-		this.constraintAdapters = new ArrayList();
-		this.selectedItems = new ArrayList();
-		this.selectionListeners = new ArrayList();
-		this.figure2ItemMap = new HashMap();
-
-		revealListeners = new ArrayList(1);
-		this.addPaintListener(new PaintListener() {
-			public void paintControl(PaintEvent e) {
-				if (!revealListeners.isEmpty()) {
-					// Go through the reveal list and let everyone know that the
-					// view is now available. Remove the listeners so they are
-					// only
-					// called once!
-					Iterator iterator = revealListeners.iterator();
-					while (iterator.hasNext()) {
-						RevealListener reveallisetner = (RevealListener) iterator.next();
-						reveallisetner.revealed(Graph.this);
-						iterator.remove();
-					}
+		this.connections = new ArrayList<>();
+		this.constraintAdapters = new ArrayList<>();
+		this.selectedItems = new ArrayList<>();
+		this.selectionListeners = new ArrayList<>();
+		this.figure2ItemMap = new HashMap<>();
+
+		revealListeners = new ArrayList<RevealListener>(1);
+		this.addPaintListener(event -> {
+			if (!revealListeners.isEmpty()) {
```

Comment:
not sure if this if is needed in modern java, the for each loop would anyhow perform the same check.